lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -641,6 +643,19 @@ bool handshake::MemoryOp::allocateMemory(
     }
   }
 
+  std::ifstream file;
+  file.open("datain" + std::to_string(id) + ".dat");
```

Comment:
This seems a bit unusual for a library to do, shouldn't the path be passed into this or something?

Reply:
Agreed that automatically doing here is a little surprising. Could this functionality be factored into some separate utility?